WORKSPACE

```diff
@@ -203,3 +203,21 @@ nodejs_register_toolchains(
 load("@bazel_skylib//:workspace.bzl", "bazel_skylib_workspace")
 
 bazel_skylib_workspace()
+
+load("@bazel_tools//tools/build_defs/repo:git.bzl", "git_repository")
+
+git_repository(
```

Comment:
In general, prefer using `http_archive` over `git_repository`. `http_archive` is much faster and doesn't require Git. You can also mirror the file and specify a `sha256` to ensure it doesn't change. I'd suggest using `https://github.com/bazelbuild/rules_dotnet/archive/refs/tags/0.0.7.zip` for the URL. Mirror to our GCS bucket (as in our main repo).

If you need to use `git_repository`, always use a specific commit, never a branch name like `master` and preferably not even a tag name like `0.0.7`: branches are expected to change. Tags may change, too.

Reply:
Okay, I have tried with `http_archive` without success. This is the error:

```
ERROR: error loading package '': Every .bzl file must have a corresponding package, but '@io_bazel_rules_dotnet//dotnet:deps.bzl' does not have one. Please create a BUILD file in the same or any parent directory. Note that this BUILD file does not need to do anything except exist.
```

I use `git_repository` and it works. Also, we should probably not mirror as this is a public example repository, it should work with standard public manners.  I am using now an specific commit.

Reply:
I'll try this out tomorrow morning and see if I can get it working with `http_archive` and without those flags.

Reply:
Did you tried @jayconrod ? If no, can we try together?

Reply:
I've had meetings this morning, but the rest of the day is open. So far, I'm running an issue with rules_kotlin in WORKSPACE. Will update you soon.

Reply:
This is the declaration I tried:

```
http_archive(
    name = "io_bazel_rules_dotnet",
    sha256 = "400416de5d5d321fa7ca9e46110373cd39db7ed84ef3a61e1150c5813cb1c99e",
    strip_prefix = "rules_dotnet-0.0.7",
    urls = ["https://github.com/bazelbuild/rules_dotnet/archive/refs/tags/0.0.7.zip"],
)
```

To prepare a declaration like this, find the URL (usually there's a release on the GitHub project, but in this case, only a tag). Download using `curl -L -O {url}`. Get the SHA-256 using `sha256sum`. These auto-generated GitHub archives put everything in a top-level directory like `rules_dotnet-0.0.7`, so you need to use `strip_prefix` to remove that.

This one didn't work for me. It has older toolchain versions (`3.1.100` is the latest platform), and the build failed. Maybe that's just me. If it doesn't work for you either, then `git_repository` is okay.

If you use `git_repository`, make sure to add `shallow_since = "1653416975 +0000"` that Bazel warns about. This will remove the warning and will make the fetch much faster.

Reply:
It works.